# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: a8b9260fbdef
# Full Hash: a8b9260fbdef0f3600dc0b8e2e245cb0f82c3470
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-09-08 21:39:05
# Regressor Bug: 1715501
# File Overlap Count: 1
# Description:
#   Bug 1715501 - patch 2 - Pass the presContext to platform font lookup methods so they can query it for font visibility. r=emilio
#   
#   This does not in itself change user-visible behavior, but is a foundation for bug
#   1715507 where we will make the behavior per-context instead of global. This is basically
#   just plumbing, passing a pointer to the presContext that's asking for fonts to be
# ==============================================================================

diff -r bed82cc6a625 -r a8b9260fbdef gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Wed Sep 08 12:18:17 2021 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Wed Sep 08 12:18:17 2021 +0000
@@ -296,6 +296,7 @@
   gfxUserFontFamily* LookupFamily(const nsACString& aName) const;
 
   virtual gfxFontSrcPrincipal* GetStandardFontLoadPrincipal() const = 0;
+  virtual nsPresContext* GetPresContext() const = 0;
 
   // check whether content policies allow the given URI to load.
   virtual bool IsFontLoadAllowed(const gfxFontFaceSrc&) = 0;